public/js/actions/pause.js

```diff
@@ -33,24 +33,19 @@ function resumed() {
  * @static
  */
 function paused(pauseInfo) {
-  return ({ dispatch, getState, client }) => {
+  return async function({ dispatch, getState, client }) {
     let { frame, frames, why } = pauseInfo;
+    frames = await updateFrameLocations(getState(), frames);
 
     dispatch(evaluateExpressions());
-
-    return dispatch({
+    dispatch({
       type: constants.PAUSED,
-      [PROMISE]: (async function () {
-        frames = await updateFrameLocations(getState(), frames);
-
-        dispatch(selectSource(frame.location.sourceId));
-        return {
-          pauseInfo: { why, frame },
-          frames: frames,
-          selectedFrameId: frame.id
-        };
-      })()
+      pauseInfo: { why, frame },
+      frames: frames,
+      selectedFrameId: frame.id
     });
+    dispatch(selectSource(frame.location.sourceId,
+                          { line: frame.location.line }));
   };
```

Comment:
I changed this to be explicit about order. I want the `selectSource` action to be fired after the pause state exists so that I can use the `selectedFrame` state in the editor. Otherwise, it will highlight that line grey for a second and then the paused state will come in and it will be highlighted green.

We never need to do any optimistic updates from the paused action anyway. Usually the `[PROMISE]` pattern is used for stuff that is user-invoked so that we can give immediate feedback.